src/AbstractRandomGenerator.ts

```diff
@@ -0,0 +1,69 @@
+const UNMISTAKABLE_CHARS =
+    '23456789ABCDEFGHJKLMNPQRSTWXYZabcdefghijkmnopqrstuvwxyz';
+const BASE64_CHARS =
+    'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ' + '0123456789-_';
+
+export default abstract class AbstractRandomGenerator {
+    /**
+     * @summary Used like `Random`, but much faster and not cryptographically secure
+     */
+    insecure!: AbstractRandomGenerator;
+    /**
+     * @summary Create a non-cryptographically secure PRNG with a given seed (using
+     * the Alea algorithm)
+     */
+    createWithSeeds!: (...seeds: any | string) => AbstractRandomGenerator;
+    /**
+     * @summary Return a number between 0 and 1, like `Math.random`.
+     */
+    abstract fraction(): number;
+
+    /**
+     * @summary Return a random string of `n` hexadecimal digits.
+     * @param digits Length of the string
+     */
+    hexString(digits: number): string {
+        return this.randomString(digits, '0123456789abcdef');
+    }
+
+    private randomString(charsCount: number, alphabet: string) {
+        let result = '';
+        for (let i = 0; i < charsCount; i++) {
+            result += this.choice(alphabet);
+        }
+        return result;
+    }
+
+    /**
+     * @summary Return a unique identifier, such as `"Jjwjg6gouWLXhMGKW"`, that is
+     * likely to be unique in the whole world.
+     * @param charsCount Optional length of the identifier in characters
+     *   (defaults to 17)
+     */
+    id(charsCount: number = 1): string {
```

Comment:
comment indicates a default value of 17, but it is set to 1

```suggestion
    id(charsCount: number = 17): string {
```

Reply:
Thanks for finding this.